packages/terra-pills/src/Private/PillsUtils.js

```diff
@@ -0,0 +1,95 @@
+const getPillWidth = (pill) => {
+  const pillStyles = window.getComputedStyle(pill);
+  const pillWidth = pill.offsetWidth + Math.round(parseFloat(pillStyles.marginLeft)) + Math.round(parseFloat(pillStyles.marginRight));
+  return pillWidth;
+};
+
+const handleRollUp = (pillListRef) => {
+  const rollUpPill = pillListRef.current.querySelector('[data-terra-rollup-pill]');
+  const containerWidth = pillListRef.current.clientWidth;
+  let rollupPillWidth;
+  if (rollUpPill) {
+    rollupPillWidth = getPillWidth(rollUpPill);
+  }
+
+  let totalPillWidth = 0;
+  let startIndex = 0;
+
+  const pills = pillListRef.current.querySelectorAll('[data-terra-pill]');
+
+  for (let i = 0; i < pills.length; i += 1) {
+    if (totalPillWidth > containerWidth) {
+      break;
+    }
+    const pillWidth = getPillWidth(pills[i]);
+    if (pillWidth + totalPillWidth < containerWidth) {
+      startIndex = i + 1;
+    } else if (rollupPillWidth + totalPillWidth > containerWidth) {
+      startIndex -= 1;
+    }
+    totalPillWidth += pillWidth;
+  }
+
+  return startIndex;
+};
+
+const getFirstFocusableNode = (pills) => {
+  let firstIndex = -1;
+  if (pills.length > 0) {
+    for (let i = 0; i < pills.length; i += 1) {
+      if ((pills[i].hasAttribute('data-terra-pill-removable')) || (pills[i].hasAttribute('data-terra-rollup-pill')) || (pills[i].hasAttribute('aria-haspopup'))) {
```

Comment:
do we need this check for roll up pill here ? as we are only trying to loop through the pills having attribute 'data-terra-pill'

Reply:
nice catch, I forgot to remove it earlier.

Removed in this commit 29ffe143cdcdef2efffe4c60df4142ad768a15d5